packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -546,8 +546,27 @@ class FlutterDriverExtension {
     final GetText getTextCommand = command as GetText;
     final Finder target = await _waitForElement(_createFinder(getTextCommand.finder));
     // TODO(yjbanov): support more ways to read text
-    final Text text = target.evaluate().single.widget as Text;
-    return GetTextResult(text.data);
+    final Widget widget = target.evaluate().single.widget;
+    switch (widget.runtimeType) {
+      case RichText:
+        final RichText richText = widget as RichText;
+        if (richText.text.runtimeType == TextSpan)
+          return GetTextResult((richText.text as TextSpan).text);
+        break;
+      case EditableText:
+        return GetTextResult((widget as EditableText).controller.text);
+      case TextField:
+        return GetTextResult((widget as TextField).controller.text);
+      case TextFormField:
+        return GetTextResult((widget as TextFormField).controller.text);
+      case Text:
+        return GetTextResult((widget as Text).data);
+      default:
+        break;
+    }
+
+    throw UnsupportedError(
+        'Type '+widget.runtimeType.toString()+' is currently not supported by getText');
```

Comment:
> PREFER using interpolation to compose strings and values.
If you’re coming from other languages, you’re used to using long chains of + to build a string out of literals and other values. That does work in Dart, but it’s almost always cleaner and shorter to use interpolation.

https://dart.dev/guides/language/effective-dart/usage#prefer-using-interpolation-to-compose-strings-and-values

Surprised this wasn't a linter error...